cdap-ui/app/cdap/components/DataPrep/index.js

```diff
@@ -45,17 +44,36 @@ export default class DataPrep extends Component {
 
     this.state = {
       backendDown: false,
-      loading: true
+      loading: true,
+      onSubmitError: null
     };
 
     this.toggleBackendDown = this.toggleBackendDown.bind(this);
     this.eventEmitter = ee(ee);
 
     this.eventEmitter.on('DATAPREP_BACKEND_DOWN', this.toggleBackendDown);
+    this.eventEmitter.on('REFRESH_DATAPREP', () => {
+      this.setState({
+        loading: true
+      });
+      /*
+        Not sure if this is necessary but added it is safer when doing an upgrade.
+        - Modified directives?
+        - Modified API calls that are not compatible with earlier version?
+      */
+      DataPrepStore.dispatch({
+        type: DataPrepActions.reset
+      });
+      setTimeout(() => {
+        this.setState({
+          loading: false
+        });
+      });
```

Comment:
Since it should just resemble a refresh I thought doing it in the next tick would be fine. From the user experience perspective it was normal too. If you feel like we need to add more time I can add it too.